hazelcast/src/main/java/com/hazelcast/map/impl/operation/PutAllBackupOperation.java

```diff
@@ -52,18 +52,18 @@ public PutAllBackupOperation() {
     @Override
     public void run() {
         boolean wanEnabled = mapContainer.isWanReplicationEnabled();
-        for (int i = 0; i < entries.size(); i++) {
-            final RecordInfo recordInfo = recordInfos.get(i);
-            final Map.Entry<Data, Data> entry = entries.get(i);
-            final Record record = recordStore.putBackup(entry.getKey(), entry.getValue());
-            Records.applyRecordInfo(record, recordInfo);
+        int i = 0;
+        for (Map.Entry<Data, Data> entry : entries) {
```

Comment:
This needs to get addressed as soon as this one is merged:
https://github.com/hazelcast/hazelcast/pull/7921